app/javascript/react/store/thresholdSlice.ts

```diff
@@ -0,0 +1,52 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+import type { RootState } from "./index";
+
+interface ThresholdState {
+  th1: number;
+  th2: number;
+  th3: number;
+  th4: number;
+  th5: number;
+}
+
+interface Threshold {
+  id: number;
+  value: number;
+}
+
+const initialState: ThresholdState = {
+  th1: 10,
+  th2: 30,
+  th3: 50,
+  th4: 80,
+  th5: 120,
+};
+
+export const thresholdSlice = createSlice({
+  name: "threshold",
+  initialState,
+  reducers: {
+    updateAll: (state, action: PayloadAction<ThresholdState>) => {
+      state.th1 = action.payload.th1;
+      state.th2 = action.payload.th2;
+      state.th3 = action.payload.th3;
+      state.th4 = action.payload.th4;
+      state.th5 = action.payload.th5;
+    },
```

Comment:
We can de-structure action payload object earlier and refer to particular `th` values when updating the state.
```suggestion
    updateAll: (
      state,
      { payload: { th1, th2, th3, th4, th5 } }: PayloadAction<ThresholdState>
    ) => {
      state.th1 = th1;
      state.th2 = th2;
      state.th3 = th3;
      state.th4 = th4;
      state.th5 = th5;
    },
```